### PR TITLE
Fix for Missile Weapons and Extendend Stats for Weapons

### PR DESCRIPTION
**Pull request by harliq**
**Missile Updates**
If tier is 4 or 5  then a chance for non elemental missile weapon with wield reqs
Tier 6 and up, Only elemental missile weapons are supposed to drop per retail.
Added 10% varaiance for Damage Modifier.  Full Modifier was rare in retail
Added +19 Elemental (like retail) and readjusted odds (odds are approximate, no hard data).  
Made MeleeDefenseMod appropiate to Bow Reqs.  Data is supports 375 wields should go upto +20, however I left them at +18 for now
Added Class TestLootGenerationFactory.cs for finding cantrips

**Added more stats for weapons in testlootgen command**
Shows Most important Modifiers/Bonuses

Example:
-----Melee Weapons----
 Wield   Damage          Variance        DefenseMod
 370     46              0.58            1.13
 370     37              0.5             1.17
 400     50              0.6             1.14
 370     37              0.35            1.13
 370     48              0.47            1.13
 400     50              0.52            1.15
 370     35              0.4             1.14
 370     57              0.45            1.19
 400     50              0.45            1.18
 370     31              0.45            1.17
 370     47              0.325           1.18
 400     43              0.4             1.11

-----Missile Weapons----
Thrown   Wield=335       Modifier=2.54   ElementalBonus=9        DefenseMod=1.1
Thrown   Wield=335       Modifier=2.51   ElementalBonus=5        DefenseMod=1.11
Thrown   Wield=375       Modifier=2.57   ElementalBonus=15       DefenseMod=1.15
Thrown   Wield=335       Modifier=2.51   ElementalBonus=7        DefenseMod=1.1
Bow      Wield=335       Modifier=2.39   ElementalBonus=6        DefenseMod=1.12
Thrown   Wield=335       Modifier=2.59   ElementalBonus=8        DefenseMod=1.14
X Bow    Wield=335       Modifier=2.56   ElementalBonus=7        DefenseMod=1.15
Bow      Wield=360       Modifier=2.36   ElementalBonus=15       DefenseMod=1.13
Bow      Wield=375       Modifier=2.34   ElementalBonus=16       DefenseMod=1.17
Thrown   Wield=360       Modifier=2.55   ElementalBonus=11       DefenseMod=1.13
Thrown   Wield=335       Modifier=2.59   ElementalBonus=7        DefenseMod=1.15

-----Caster Weapons----
Wield=150        ElementalDamageMod= 0.00        DefenseMod=1.15
Wield=375        ElementalDamageMod=1.15         DefenseMod=1.18
Wield=355        ElementalDamageMod=1.15         DefenseMod=1.15
Wield=355        ElementalDamageMod=1.15         DefenseMod=1.15
Wield=150        ElementalDamageMod= 0.00        DefenseMod=1.15
Wield=355        ElementalDamageMod=1.15         DefenseMod=1
Wield=150        ElementalDamageMod= 0.00        DefenseMod=1.18

### CODE REVIEW

**Comment by ziang4891** [comment]
I would suggest moving GetBowMeleeDMod() to LootGenerationFactory_Missile.cs, as that method is related only to missile weapons

File: Source/ACE.Server/Factories/LootGenerationFactory.cs [hunk 1]
```diff
@@ -2249,5 +2249,106 @@ private static float GetTotalProbability(List<TreasureMaterialGroups> list)
...
+                        meleeMod = 0.13;
+                    else if (chance < 80)
+                        meleeMod = 0.14;
+                    else if (chance < 95)
+                        meleeMod = 0.15;
+                    else
+                        meleeMod = 0.10;
+                    break;
+                case 360:
+                case 375:
```
Comment (ziang4891) [inline Source/ACE.Server/Factories/LootGenerationFactory.cs]: Missing entry for wield 385

Comment (harliq) [reply Source/ACE.Server/Factories/LootGenerationFactory.cs]: Fixed

**Comment by ziang4891** [comment]
Successfully tested on my private server

**Comment by harliq** [comment]
Pulling out TestLootGen Update, will open a new PR

**Pull request closed without merging**